encoders/csv.go

```diff
@@ -1,11 +1,50 @@
 package encoders
 
-import "github.com/jszwec/csvutil"
+import (
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/jszwec/csvutil"
+	"io"
+)
+
+var _ runtime.Marshaler = (*csvEncoder)(nil)
 
 // csvEncoder implements Marshaller for the CSV format.
 type csvEncoder struct {
 }
 
+// NewDecoder returns a Decoder which reads byte sequence from "r".
+func (e csvEncoder) NewDecoder(r io.Reader) runtime.Decoder {
+	return runtime.DecoderFunc(func(v interface{}) error {
+		raw, err := io.ReadAll(r)
+		if err != nil {
+			return err
+		}
+		return e.Unmarshal(raw, v)
+	})
+}
+
+// NewEncoder returns an Encoder which writes bytes sequence into "w".
+func (e csvEncoder) NewEncoder(w io.Writer) runtime.Encoder {
+	return runtime.EncoderFunc(func(v interface{}) error {
+		b, err := e.Marshal(v)
+		if err != nil {
+			return err
+		}
+		_, err = w.Write(b)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
```

Comment:
This same exact implementation is shared by the `raw` implementation, and I'm sure it will be shared by others as the steps are fairly generic. Let's extract this and reuse it across implementations.